node/node.go

```diff
@@ -597,11 +606,25 @@ func (node *Node) Start() error {
 						handleEArg:     openBox,
 					}
 				default:
+					errChan <- withError{
+						errors.WithStack(errUnkonwnP2PMessageType),
+						proto.MessageCategory(openBox[proto.MessageCategoryBytes-1]),
+					}
 					return false
 				}
 
-				return true
+				select {
+				case <-ctx.Done():
+					if errors.Is(ctx.Err(), context.DeadlineExceeded) {
+						utils.Logger().Warn().
+							Str("topic", topicNamed).Msg("[context] exceeded validation deadline")
+					}
+					errChan <- withError{errors.WithStack(ctx.Err()), msg}
+				default:
+					return true
+				}
 
+				return true
```

Comment:
Shall return false if timeout right?

Reply:
make sense.  I need to test this scenario though.